src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -530,14 +530,23 @@ public void startJenkins() throws Throwable {
                 "--httpPort=" + port, // initially port=0. On subsequent runs, the port is set to the port used allocated randomly on the first run.
                 "--httpListenAddress=127.0.0.1",
                 "--prefix=/jenkins"));
-        ProcessBuilder pb = new ProcessBuilder(argv);
-        System.out.println("Launching: " + pb.command());
-        pb.environment().put("JENKINS_HOME", home.getAbsolutePath());
+        Map<String, String> env = new TreeMap<>();
```

Comment:
Nice touch to use something better than `HashMap` here (and consistent with `hudson.EnvVars`, which also uses `TreeMap`). It's too bad that `ProcessBuilder#environment` uses a `HashMap` internally, scrambling the order of whatever the user passes in. I wish `hudson.EnvVars` had extended `LinkedHashMap` (insertion order), because that is how libc's `setenv(3)` works, but that ship sailed a long time ago.

Reply:
As far as I know the ultimate order is irrelevant, but it is much nicer to print the variables in order below.

Reply:
> but it is much nicer to print the variables in order below.

… which is why I wrote it was a nice touch above.